src/fstext/fstext-utils-inl.h

```diff
@@ -374,6 +374,16 @@ void GetSymbols(const SymbolTable &symtab,
                 std::vector<I> *syms_out) {
   KALDI_ASSERT(syms_out != NULL);
   syms_out->clear();
+#if OPENFST_VER >= 10800
+  for (SymbolTable::iterator iter = symtab.begin();
+      iter != symtab.end();
+      ++iter) {
```

Comment:
better to use auto with range based for here.

Reply:
You can argue with @danpovey about this.🙂 

In this particular case, I agree with you. My own rule of thumb is, if the _real_ or _conceptual_ type is obvious, use either `auto`; otherwise, the type must be explicit. In this case, the conceptual type, namely the iterator, is obvious. But Dan wants explicit types, with an exception for extreme ugliness of their explicit spelling, I believe.